fitlins/cli/run.py

```diff
@@ -170,18 +168,53 @@ def run_fitlins(argv=None):
         pipeline_name += '_' + opts.derivative_label
     deriv_dir = op.join(opts.output_dir, pipeline_name)
     os.makedirs(deriv_dir, exist_ok=True)
-
     bids.write_derivative_description(opts.bids_dir, deriv_dir)
 
     work_dir = mkdtemp() if opts.work_dir is None else opts.work_dir
 
+    # Go ahead and initialize the layout database
+    if opts.database_path is None:
+        database_path = Path(work_dir) / 'dbcache'
+        reset_database = True
```

Comment:
Do we need to manually ask pybids to `reset_database`? It will do it on its own if it doesn't file a valid `sqlite` file inside the `database_path`. 

I suppose this is safer though, as it could be a mismatching `dbcache` from another run. 

An alternative here is to clean up the `dbcache` folder on fitlins teardown. (and put it somewhre more hidden, at least make it `.dbcache`). Or use a tmpdir, although @effigies seemed to think that would be a problem (not guaranted to be on disk).

I think the right solution is to use `tempfile.TemporaryDirectory` but set `prefix` to `work_dir/.dbcache`. That way its cleaned up automatically. Or can also  be cleaned up manually with ``cleanup` function.

Reply:
> I suppose this is safer though, as it could be a mismatching `dbcache` from another run.

Yes, that's the purpose here. If somebody passes a db file, then we assume they know that it's valid. If we're creating it, we don't want to take any chances, and will always clear it.



> I think the right solution is to use `tempfile.TemporaryDirectory` but set `prefix` to `work_dir/.dbcache`. That way its cleaned up automatically.

That would be fine, but I tend to think of anything in the scratch directory as disposable anyway.

Reply:
That's true. Maybe I'm overthinking this then, and it might actually be useful to have the index there for debugging. 

I suppose there could be an advantage to not re-indexing if you're running fitlins over and over. Maybe we can over-ride the force-re-index behavior for power users?

Reply:
I deffo want it for debugging, and this way, if they want to reuse a layout saved in a working directory, they can pass the path for it explicitly, otherwise it gets reset in order to avoid nasty surprises.